src/main/java/health/ere/ps/service/connector/endpoints/EndpointDiscoveryService.java

```diff
@@ -0,0 +1,201 @@
+package health.ere.ps.service.connector.endpoints;
+
+import health.ere.ps.exception.common.security.SecretsManagerException;
+import health.ere.ps.service.common.security.SecretsManagerService;
+import health.ere.ps.ssl.SSLUtilities;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Invocation;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This service automatically discovers the endpoints that are available at the connector.
+ */
+@ApplicationScoped
+public class EndpointDiscoveryService {
+    private static final Logger LOG = Logger.getLogger(EndpointDiscoveryService.class.getName());
+
+    /**
+     * Certificate to authenticate at the connector.
+     */
+    @ConfigProperty(name = "connector.cert.auth.store.file")
```

Comment:
In #32 it was priotized to have all @configProperty at one place to better manage the configurations at one place.
It is better so, that refactoring and managment is to minimize. Can you leave the properties in AppConfig.

I personal think the better way to bind properties are `@ConfigProperties`, because it removes the disadvantage of `@ConfigProperty`.

Reply:
The priority of #32 was to eliminate dupplicates.

Reply:
This service determines the endpoint addresses such that they may not stay in `AppConfig`.